app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayController.kt

```diff
@@ -144,18 +159,41 @@ class DefaultTabsTrayController(
      * Deletes the [TabSessionState] with the specified [tabId].
      *
      * @param tabId The id of the [TabSessionState] to be removed from TabsTray.
+     * @param source app feature from which the tab with [tabId] was closed.
      * This method has no effect if the tab does not exist.
      */
-    override fun handleTabDeletion(tabId: String) {
+    override fun handleTabDeletion(tabId: String, source: String?) {
+        deleteTab(tabId, source, isConfirmed = false)
+    }
+
+    override fun handleDeleteTabWarningAccepted(tabId: String, source: String?) {
+        deleteTab(tabId, source, isConfirmed = true)
+    }
+
+    private fun deleteTab(tabId: String, source: String?, isConfirmed: Boolean) {
         val tab = browserStore.state.findTab(tabId)
 
         tab?.let {
-            if (browserStore.state.getNormalOrPrivateTabs(it.content.private).size != 1) {
+            val isLastTab = browserStore.state.getNormalOrPrivateTabs(it.content.private).size == 1
+            if (!isLastTab) {
                 tabsUseCases.removeTab(tabId)
                 showUndoSnackbarForTab(it.content.private)
             } else {
-                dismissTabsTrayAndNavigateHome(tabId)
+                val privateDownloads = browserStore.state.downloads.filter { map ->
+                    map.value.private && (
+                        map.value.status == DownloadState.Status.INITIATED ||
+                            map.value.status == DownloadState.Status.DOWNLOADING ||
+                            map.value.status == DownloadState.Status.PAUSED
+                        )
+                }
```

Comment:
Maybe we could have a extension function to reuse this code (arePrivateOngoingDownloads, it looks the same as what we have here https://github.com/mozilla-mobile/fenix/pull/22912/files#diff-5912831edfde6ad16931cf998b691e99e6e5c81d6e8887b73a0421efc842f272R185-R191